### PR TITLE
[openstack_ceilometer] Add backend_url to protected keys

### PR DESCRIPTION
**Pull request by jcastill**
This patch adds the password and url in backend_url to the
list of protected keys, so it gets properly masked.

Related: RHBZ#1892692
Resolves: #2298 

Signed-off-by: Jose Castillo <jose.mfcastillo@gmail.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [] If this commit closes an existing issue, is the line `Closes: #ISSUENUMBER` included in an independent line?
- [X] If this commit resolves an existing pull request, is the line `Resolves: #PRNUMBER` included in an independent line?

### CODE REVIEW

**Review by pmoravec — COMMENTED** [review]
For both `transpert_url` and `backend_url`, I think there is no need to obfuscate hostname and URI. If am I right, isn't it worth moving them from `protect_keys` to `connection_keys`?

**Comment by jcastill** [comment]
I was in doubt between these two, but what made me decide for protected_keys was that transport_url was added there. That said, the hostname/IP can be an internal customer's IP, so I thought having it masked as well was a good idea. But I can change backend_url (and transport_url as well?) to connection_keys if you prefer

**Comment by TurboTurtle** [comment]
I think we should rely on `--clean` to obfuscate the hostnames here if it is desired by the user.

**Comment by jcastill** [comment]
ok, I'll move backend_url to connection_keys. What about transport_url?

**Review by pmoravec — APPROVED** [review]
IMHO it would make sense to move `transport_url` to `connection_keys` as well, but a review from OSP is needed (if the format of the obfuscated data match).

ACK to this change itself.

**Comment by TurboTurtle** [comment]
@jcastill this looks ready to merge once `transport_url` is added to `connection_keys`.

**Review by pmoravec — APPROVED** [review]

**Comment by pmoravec** [comment]
The packit failure seems an infra problem not on our side, we can ignore it.

**Review by TurboTurtle — APPROVED** [review]
Yeah, I'm disabling the packit service again until we get a better way to auth builds for PRs from contributors without write access to the repo.

This LGTM as well.

**Pull request closed without merging**
